### PR TITLE
AV-197542 update alb-sdk and ako

### PR DESCRIPTION
**Pull request by arihantg**
This Pr upgrades alb-sdk to 30-2-1 and corresponding ako library. 
IT also removes the hardcoded api version of 22.1.2 and introduces pkiProfile reference in GDP and GSLBHostrule CRD.

### CODE REVIEW

**Comment by arihantg** [comment]
build amko

**Comment by arihantg** [comment]
build amko

**Comment by arihantg** [comment]
build amko

**Comment by arihantg** [comment]
build amko

**Comment by arihantg** [comment]
build amko

File: helm/amko/crds/gdp_def.yaml [hunk 1]
```diff
@@ -64,6 +64,8 @@ spec:
                 maximum: 86400
               sitePersistenceRef:
                 type: string
+              pkiProfileRef:
```
Comment (akshayhavile) [inline helm/amko/crds/gdp_def.yaml:67]: will this be applied to gslb service only when sitePersistence is set in gslbhostrule? will this take precedence over gslbhostrule pkiProfileRef?

**Review by akshayhavile — COMMENTED** [review]
Have we tested this changes in older controller versions?

File: helm/amko/crds/gdp_def.yaml (hunk 1 above)
Comment (arihantg) [reply helm/amko/crds/gdp_def.yaml:67]: We follow the same rules as for persistence profile. GSLB Hostrule always take precedence over GDP 
  1. GSLBHostRule contains site persistence, but it is disabled:
    Regardless of what's in the GDP object, we disable Site Persistence on this GS object so no pki Profile will be set.  2. GSLBHostRule contains site persistence, it is enabled and a pki profile ref is given:
   We enable Site Persistence on the GS object and set the provided ref as the pki ref.
 3. GSLBHostRule doesn't contain PKI, but site persistence is enabled then we inherit the PKi profile from
   the Global filter (GDP object).
   4.GSLBHostrule does not contain site persistence object. We rely on fields in GDP object.

File: gslb/rest/dq_nodes.go [hunk 2]
```diff
@@ -1071,17 +1075,23 @@ func (restOp *RestOperations) AviGSBuild(gsMeta *nodes.AviGSObjectGraph, restMet
...
 	if gsMeta.SitePersistenceRef != nil {
 		sitePersistenceEnabled := true
 		persistenceProfileRef := "/api/applicationpersistenceprofile?name=" + *gsMeta.SitePersistenceRef
 		aviGslbSvc.SitePersistenceEnabled = &sitePersistenceEnabled
 		aviGslbSvc.ApplicationPersistenceProfileRef = &persistenceProfileRef
+		if gsMeta.PkiProfileRef != nil {
+			pkiProfileRef := "/api/pkiprofile?name=" + *gsMeta.PkiProfileRef
+			aviGslbSvc.PkiProfileRef = &pkiProfileRef
+		} else {
+			aviGslbSvc.PkiProfileRef = gslbutils.GetDefaultPKI(cache.SharedAviClients().AviClient[0])
```
[the remark this answers is not in the document]
Comment (arihantg) [reply gslb/rest/dq_nodes.go:1093]: It is expected from 22.1.3. Before that it was set by controller in background and not visible in GSLBservice.

File: gslb/nodes/node_utils.go [hunk 3]
```diff
@@ -43,6 +43,18 @@ func getSitePersistence(gsRuleExists bool, gsRule *gslbutils.GSHostRules, gf *gs
 	return gf.GetSitePersistence()
 }
 
+func getPKIProfile(gsRuleExists bool, gsRule *gslbutils.GSHostRules, gf *gslbutils.GlobalFilter) *string {
+	if gsRuleExists && gsRule.SitePersistence != nil && gsRule.SitePersistence.PKIProfileRef != nil {
+		if gsRule.SitePersistence.Enabled {
+			ref := gsRule.SitePersistence.PKIProfileRef
+			return ref
+		} else {
```
[the remark this answers is not in the document]
Comment (arihantg) [reply gslb/nodes/node_utils.go:51]: else is needed to return nil if site Persistence is disabled but pkiprofile is still present

File: gslb/ingestion/gslb.go [hunk 4]
```diff
@@ -494,13 +494,13 @@ func parseControllerDetails(gc *gslbalphav1.GSLBConfig) error {
...
 	ctrlUsername := secretObj.Data["username"]
 	ctrlPassword := secretObj.Data["password"]
-
-	// Setting the client version as 22.1.2 as a work around for the GSLB service creation failure
-	// in 22.1.3 due to introduction of mandatory PKI profile.
-	// TODO: Change this logic and incorporate the PKI profile in the future releases.
-	if leaderVersion > gslbutils.CONTROLLER_VERSION_22_1_2 {
-		leaderVersion = gslbutils.CONTROLLER_VERSION_22_1_2
-	}
+	/*
```
[the remark this answers is not in the document]
Comment (arihantg) [reply gslb/ingestion/gslb.go:497]: will remove

File: gslb/apiserver/ingestion.go [hunk 5]
```diff
@@ -92,7 +93,7 @@ type AcceptedIngressAPI struct{}
 
 func (ai AcceptedIngressAPI) InitModel() {}
 
-func (ai AcceptedIngressAPI) ApiOperationMap() []models.OperationMap {
+func (ai AcceptedIngressAPI) ApiOperationMap(prometheusEnavbled bool, reg *prometheus.Registry) []models.OperationMap {
```
[the remark this answers is not in the document]
Comment (arihantg) [reply gslb/apiserver/ingestion.go:96]: yes there were compilation issues

File: gslb/rest/dq_nodes.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (arihantg) [reply gslb/rest/dq_nodes.go:1093]: similar  migration routine added to migrate to 22.1.3 controller 
https://github.com/avinetworks/avi-dev/pull/106542/files

File: gslb/test/ingestion/gslbhostrule_test.go [hunk 6]
```diff
@@ -103,6 +103,39 @@ func TestGSLBHostRuleInvalidSitePersistence(t *testing.T) {
 	t.Logf("Verified GSLBHostRule")
 }
 
+func TestGSLBHostRuleValidPkiProfile(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (arihantg) [reply gslb/test/ingestion/gslbhostrule_test.go:106]: working on it. Will add in another PR if it can get integrated

**Comment by arihantg** [comment]
> Have we tested this changes in older controller versions?

only tested in 30.2.1 so far

**Comment by arihantg** [comment]
build amko

File: gslb/gslbutils/constants.go [hunk 7]
```diff
@@ -127,4 +127,7 @@ const (
 
 	// Controller version constants
 	CONTROLLER_VERSION_22_1_2 = "22.1.2"
```
Comment (akshayhavile) [inline gslb/gslbutils/constants.go:129]: Can we take out this: CONTROLLER_VERSION_22_1_2?

**Comment by arihantg** [comment]
> > Have we tested this changes in older controller versions?
> 
> only tested in 30.2.1 so far

tested GSLB creation with 22.1.3 as well manually

**Comment by arihantg** [comment]
build amko

**Review by akshayhavile — APPROVED** [review]

**Review by DixitAakash — APPROVED** [review]
Overall changes look good.

**Pull request merged**